src/engine/Join.cpp

```diff
@@ -172,6 +172,24 @@ Result Join::computeResult(bool requestLaziness) {
   return lazyJoin(std::move(leftRes), std::move(rightRes), requestLaziness);
 }
 
+// _____________________________________________________________________________
+std::optional<std::shared_ptr<QueryExecutionTree>>
+Join::setPrefilterGetUpdatedQueryExecutionTree(
+    std::vector<PrefilterVariablePair> prefilterVariablePairs) const {
```

Comment:
I think we shouldn't copy the vector each time.
If a operation actually succeeds to use a prefilter and therefore needs a copy to store it, then it can make that copy itself. 
Copying just for traversing and looking is wasteful.